src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -202,10 +203,16 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         .select(
           allIndexColumns.head,
           allIndexColumns.tail :+ IndexConstants.DATA_FILE_NAME_ID: _*)
+        .toDF(SchemaUtils.prefixNestedFieldNames(allIndexColumns) :+
```

Comment:
We do this to transform the selected fields into the prefixed nested ones.

Reply:
Not directly. `allIndexColumns != escapedIndexedColumns ++ escapedIncludedColumns`. The `allIndexColumns` contains the columns contained in the index and missing columns.

I will replace it with `escapedIndexedColumns ++ escapedIncludedColumns ++ missingPartitionColumns` which is longer but is ok.